src/monodroid/jni/android-system.h

```diff
@@ -19,22 +20,22 @@ namespace xamarin { namespace android {
 
 namespace xamarin { namespace android { namespace internal
 {
+#if defined (DEBUG) || !defined (ANDROID)
 	struct BundledProperty;
-
-	struct KnownEnvironmentVariables
-	{
-		static constexpr bool AssemblyPreloadDefault = true;
-
-		bool        DSOInApk = false;
-		MonoAotMode MonoAOT = MonoAotMode::MONO_AOT_MODE_NONE;
-		bool        MonoLLVM = false;
-		bool        EnableAssemblyPreload = AssemblyPreloadDefault;
-	};
+#endif
 
 	class AndroidSystem
 	{
+		// These values are completely arbitrary, but they should be quite enough for your
+		// purposes without being too limiting
+		static constexpr unsigned long MAX_ENVVAR_NAME_WIDTH = 512;
+		static constexpr unsigned long MAX_ENVVAR_VALUE_WIDTH = 1024;
```

Comment:
Throw in some `MONO_`-prefixed environment variables, and we very well may be able to exceed that...